crates/typst/src/math/equation.rs

```diff
@@ -200,16 +229,7 @@ impl Layout for EquationElem {
                 frame.push_frame(Point::new(x, y), counter)
             }
         } else {
-            let font_size = TextElem::size_in(styles);
-            let slack = ParElem::leading_in(styles) * 0.7;
-            let top_edge = TextElem::top_edge_in(styles).resolve(font_size, &font, None);
-            let bottom_edge =
-                -TextElem::bottom_edge_in(styles).resolve(font_size, &font, None);
-
-            let ascent = top_edge.max(frame.ascent() - slack);
-            let descent = bottom_edge.max(frame.descent() - slack);
-            frame.translate(Point::with_y(ascent - frame.baseline()));
-            frame.size_mut().y = ascent + descent;
+            adjust_for_leading(&mut frame, &font, styles);
```

Comment:
I wonder whether this `else` is dead code, since the whole `layout` call should only happen for block equations.